agent/providers/api.go

```diff
@@ -27,7 +28,7 @@ type ExecutionProvider interface {
 	Deploy() error
 
 	// Execute a deployed function, if supported by the execution provider implementation (e.g., "v8" and "wasm" types)
-	Execute(subject string, payload []byte) ([]byte, error)
+	Execute(headers nats.Header, payload []byte) ([]byte, error)
```

Comment:
let's make this a `context.Context` here, as is the Go convention. We shouldn't be bleeding the nats.Header type into the interface for a host services provider client.

Reply:
we can use `context.WithValue(parent_ctx, key, value)` to produce a context that has the information the provider needs, like the trigger subject that originally instigated the call.

https://pkg.go.dev/context#WithValue